core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -872,11 +872,12 @@ final class Stream[+F[_], +O] private (private val free: FreeC[Algebra[Nothing,
     }.stream
 
   /**
-    * Tracks any resources acquired during this stream and releases them when the stream completes.
+    * Scopes are typically inserted automatically, at the boundary of a pull (i.e., when a pull
+    * is converted to a stream). This method allows a scope to be explicitly demarcated so that
+    * resources can be freed earlier than when using automatically inserted scopes.
     *
-    * Scopes are sometimes inserted automatically, (e.g., as a result of calling `handleErrorWith`).
-    * This method allows a scope to be explicitly demarcated, so that resources can be freed earlier
-    * than when using automatically inserted scopes.
+    * One use case is scoping the left hand side of an append: `(s1.scope ++ s2)`, which ensures
```

Comment:
A bit confused if this is the case now or not.
If I don't insert `scope` explicitly, are resources in `s1 ++ s2` released at the end of `s1` or not?

Reply:
Yeah, the comment is pretty weak. If `s1` reaches a natural termination point, then it's resources are released before moving on to `s2`. However, if it doesn't run to completion, e.g. because of a pull that discarded the tail and used `streamNoScope`, then the explicit resource release call is lost and the resource lifecycle ends up promoted to the lifecycle of the active scope.

I don't know how to document this but suggestions are welcome. :)

Reply:
> However, if it doesn't run to completion, e.g. because of a pull that discarded the tail and used streamNoScope, then the explicit resource release call is lost

That's only true if `streamNoScope` is used though, right? (and it's true currently, which is the source of the leaks?)
Or can it also happen to a Stream that didn't run to completion but didn't use `streamNoScope`?

Reply:
I'm being picky because:
- That's one of the few comments addressing `Scopes`, and it's confusing :) If we can't make it clearer (and I agree that it's hard to document), we might be better off removing it
- Given the findings in #1151 , perhaps `streamNoScope` should come with a "only use it if you know what you're doing" warning

Reply:
Agreed on both counts. Will push an update now.